### PR TITLE
User/akshay/add status to game state

### PR DESCRIPTION
**Pull request by gulatiak-usc**
Added status tracking in game state
Connected set_status/remove_status to BotClient API for DQN training

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -351,35 +351,48 @@ def set_and_check_team(self, player, team):
 				status = params[1]
 				if self.own_pokemon(pokemon_data):
 					self.add_status(self.statuses, pokemon_name, status)
```
Comment (trevor-e-apple) [inline bot.py]: It would probably be more concise to just move calls of self.gs.set_status to self.add_status, right?

File: bot.py [hunk 2]
```diff
@@ -351,35 +351,48 @@ def set_and_check_team(self, player, team):
...
 				else:
 					self.add_status(self.opp_statuses, pokemon_name, status)
+					self.gs.set_status(GameState.Player.two, pokemon_name, status)
 
 			elif inp_type == '-end':
 				pokemon_data = params[0]
 				pokemon_name = self.get_pokemon(pokemon_data)
 				status = params[1]
 				if self.own_pokemon(pokemon_data):
 					self.remove_status(self.statuses, pokemon_name, status)
```
Comment (trevor-e-apple) [inline bot.py]: similar thing here, self.gs.remove_status could probably just be a part of self.remove_status

File: gamestate.py [hunk 3]
```diff
@@ -389,6 +451,16 @@ def test_types(player, pokemon, type_names):
 			gs.set_type(player, pokemon, type_name)
 		check_types(player, pokemon, type_names)
 	
+	def check_statuses(player, pokemon, status_names):
```
Comment (trevor-e-apple) [inline gamestate.py]: These test functions never actually seem to be used.

**Review by trevor-e-apple — CHANGES_REQUESTED** [review]
I think the test code needs to be used and run for gamestate.py. I think the changes to bot.py that I suggested would probably help with the maintainability of the code.

File: bot.py (hunk 1 above)
Comment (gulatiak-usc) [reply bot.py]: Makes sense

File: bot.py (hunk 2 above)
Comment (gulatiak-usc) [reply bot.py]: Makes sense

File: gamestate.py (hunk 3 above)
Comment (gulatiak-usc) [reply gamestate.py]: Correct! Working on it

Comment (gulatiak-usc) [reply gamestate.py]: Resolved

**Comment by trevor-e-apple** [comment]
Hm, this branch somehow broke active tracking which works fine on master, and remove statuses doesn't seem to work well either. I'm doing a little investigation on my side.

**Pull request closed without merging**